pkg/deploy/generator/resources_rp.go

```diff
@@ -1095,10 +1099,14 @@ func (g *generator) database(databaseName string, addDependsOn bool) []*arm.Reso
 		},
 	}
 
-	if g.production {
-		database.Resource.(*mgmtdocumentdb.SQLDatabaseCreateUpdateParameters).SQLDatabaseCreateUpdateProperties.Options.Throughput = to.Int32Ptr(cosmosDbStandardProvisionedThroughputHack)
-		portal.Resource.(*mgmtdocumentdb.SQLContainerCreateUpdateParameters).SQLContainerCreateUpdateProperties.Options.Throughput = to.Int32Ptr(cosmosDbPortalProvisionedThroughputHack)
-		gateway.Resource.(*mgmtdocumentdb.SQLContainerCreateUpdateParameters).SQLContainerCreateUpdateProperties.Options.Throughput = to.Int32Ptr(cosmosDbGatewayProvisionedThroughputHack)
+	if !g.production {
+		database.Resource.(*mgmtdocumentdb.SQLDatabaseCreateUpdateParameters).SQLDatabaseCreateUpdateProperties.Options = &mgmtdocumentdb.CreateUpdateOptions{
+			AutoscaleSettings: &mgmtdocumentdb.AutoscaleSettings{
+				MaxThroughput: to.Int32Ptr(500),
```

Comment:
It looks like the most recent E2E test run failed because 1000 is the minimum allowed value here. See https://learn.microsoft.com/en-us/azure/cosmos-db/provision-throughput-autoscale#how-autoscale-provisioned-throughput-works, which lines up with the error message.

Reply:
Good catch, thanks for the heads-up.